src/app/shared/form/form-controls-types/form-controls-text.ts

```diff
@@ -0,0 +1,11 @@
+import { FormControlsBase } from './form-controls-base';
+
+export class FormControlsText extends FormControlsBase<string> {
+  controlType = 'text';
+  type: string;
+
+  constructor(options: {} = {}) {
```

Comment:
Not going to hold up your PR, but this and the textarea type seem identical, so not sure why it's not on the base class.

(Btw, this is why I've grown to detest classical OOP. The ritual is exhausting.)